test/groovy/templates/PiperPipelineStageComplianceTest.groovy

```diff
@@ -20,22 +21,29 @@ class PiperPipelineStageComplianceTest extends BasePiperTest {
         .around(jlr)
         .around(jsr)
 
+    private List stepsCalled = []
+    private Map stepParameters = [:]
+
     @Before
     void init()  {
         binding.variables.env.STAGE_NAME = 'Compliance'
         helper.registerAllowedMethod('piperStageWrapper', [Map.class, Closure.class], {m, body ->
             return body()
         })
+        helper.registerAllowedMethod('sonarExecuteScan', [Map.class], {m ->
+            stepsCalled.add('sonarExecuteScan')
+            stepParameters.sonarExecuteScan = m
+        })
     }
 
     @Test
     void testStageDefault() {
-
         jsr.step.piperPipelineStageCompliance(
             script: nullScript,
             juStabUtils: utils,
+            sonarExecuteScan: true
```

Comment:
I think in the default test case you should test that no step is running by default, i.e., in case you do not set any step to true. And then have another test case where you activate the step and check that it was executed. At least that is the test case pattern used by other stages.